sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2Suites.scala

```diff
@@ -635,6 +683,7 @@ abstract class HiveThriftServer2Test extends SparkFunSuite with BeforeAndAfterAl
        |  --driver-class-path $driverClassPath
        |  --driver-java-options -Dlog4j.debug
        |  --conf spark.ui.enabled=false
+       |  ${extraConf.mkString("\n")}
```

Comment:
Do you mean something like `--single-session`? The benefit of `SparkConf` option is that users can add it in spark-defaults.conf.

Command line options of `start-thriftserver.sh` are parsed by both `SparkSubmit` and `HiveServer2.ServerOptionsProcessor`. The latter isn't extensible. One option is to add a `--single-session` option in `SparkSubmit` and make a synonymous of `--conf spark.sql.hive.thriftServer.singleSession=true`. However, this option doesn't seem to be general enough to be added in `SparkSubmit`.